test/decl/typealias/generic.swift

```diff
@@ -237,16 +253,13 @@ let _: ConcreteStruct.O<Int> = ConcreteStruct.O<Int>(123)
 
 // Qualified lookup of generic typealiases nested inside generic contexts
 //
-// FIXME marks cases which still don't work correctly, and either produce a
-// spurious diagnostic, or are actually invalid and do not diagnose.
-//
 // This occurs because the constraint solver does the wrong thing with an
 // UnresolvedSpecializeExpr applied to a generic typealias.
 //
 // In the other cases, we manage to fold the UnresolvedSpecializeExpr in the
 // precheckExpression() phase, which handles generic typealiases correctly.
 
-let _ = GenericClass.TA<Float>(a: 4.0, b: 1) // FIXME
```

Comment:
I think these might still be wrong? Can you verify that it produces the correct underlying type vs when the alias appears in type context?

Reply:
Oh, good catch, you’re right! What an odd bug though; I will bring back the FIXMEs and add a verification of this error.

```swift
struct S<X, Y> {
  var x: X, y: Y
}
class C<T> {
  typealias TA<U> = S<T, U>
}

let s1 = C.TA<Float>(x: 4.0, y: 1)
let s2: S<Float, Float> = s1 // cannot assign value of type 'S<Float, Int>' to type 'S<Float, Float>'
```

Reply:
I believe the bug is because SpecializeExpr doesn't handle type alias types correctly. It just desugars the underlying type.